### ISSUE TITLE
No change in AR binary tag segmentation in TECA-BARD

### ISSUE THREAD

**Issue by VThandlam** [LBL-EESA/TECA#791]
Hi,

I see no change in the AR binary tag segmentation mapping despite running the TECA-BARD with different segment_threshold values. Attached is the figure generated from two such runs. The filled area is the output with segement_threshold 1, and the blue contour is the one generated using the default value of 0.6667.  You can find the command lines I have used below.

teca_bayesian_ar_detect --input_regex 20201era5.nc --x_axis_variable longitude --y_axis_variable latitude --specific_humidity q --wind_u u --wind_v v --ivt_u ivt_u --ivt_v ivt_v --ivt ivt  --compute_ivt --write_ivt --write_ivt_magnitude --n_threads -1 --verbose --z_axis_variable level --segment_ar_probability -segment_threshold 0.6667 -segment_variable ar_binary_tag --output_file 20201era5_BARD_std_%t%.nc

teca_bayesian_ar_detect --input_regex 20201era5.nc --x_axis_variable longitude --y_axis_variable latitude --specific_humidity q --wind_u u --wind_v v --ivt_u ivt_u --ivt_v ivt_v --ivt ivt  --compute_ivt --write_ivt --write_ivt_magnitude --n_threads -1 --verbose --z_axis_variable level --segment_ar_probability -segment_threshold 1 -segment_variable ar_binary_tag --output_file 20201era5_BARD_st1_%t%.nc

[seg_diff.pdf](https://github.com/LBL-EESA/TECA/files/13624488/seg_diff.pdf)

**Comment by taobrienlbl** [comment]
Thanks for submitting this @venugopalrt - I'll take a look this week

**Comment by VThandlam** [comment]
Hi again,

Below are the warning messages I am getting while running the teca bard.  
WARNING: [0:140682412433024] [/home/venu/TECA/io/teca_cf_layout_manager.cxx:375 4.0.0-247-gb06b982]
WARNING: attributes for "ar_probability" are missing the mesh_dim_active key. All mesh dimensions are assumed to be active.
WARNING: [0:140682412433024] [/home/venu/TECA/io/teca_cf_layout_manager.cxx:375 4.0.0-247-gb06b982]
WARNING: attributes for "ivt" are missing the mesh_dim_active key. All mesh dimensions are assumed to be active.
WARNING: [0:140682412433024] [/home/venu/TECA/io/teca_cf_layout_manager.cxx:375 4.0.0-247-gb06b982]
WARNING: attributes for "ivt_u" are missing the mesh_dim_active key. All mesh dimensions are assumed to be active.
WARNING: [0:140682412433024] [/home/venu/TECA/io/teca_cf_layout_manager.cxx:375 4.0.0-247-gb06b982]
WARNING: attributes for "ivt_v" are missing the mesh_dim_active key. All mesh dimensions are assumed to be active.
WARNING: [0:140682412433024] [/home/venu/TECA/io/teca_cf_layout_manager.cxx:375 4.0.0-247-gb06b982]
WARNING: attributes for "ar_binary_tag" are missing the mesh_dim_active key. All mesh dimensions are assumed to be active.

**Comment by taobrienlbl** [comment]
Hi @venugopalrt - apologies it took longer than expected to get to this.  We're down a team member, and I've been spread pretty thin.

I just tried to replicate the problem, and I'm unable to.

Here's the output for a threshold of 0.1:
```
teca_bayesian_ar_detect \
  --input_file teca_era5_debug_dataset.mcf \
  --compute_ivt_magnitude \
  --ivt_u="VIWVE" \
  --ivt_v="VIWVN" \
  --segment_threshold=0.1 \
  --segment_ar_probability
```
![Screenshot 2024-01-25 155818](https://github.com/LBL-EESA/TECA/assets/8796694/08d056d2-5cf4-4ad1-91e7-2cc1085801b6)

And here's the output with a threshold of 0.9:
```
teca_bayesian_ar_detect \
  --input_file teca_era5_debug_dataset.mcf \
  --compute_ivt_magnitude \
  --ivt_u="VIWVE" \
  --ivt_v="VIWVN" \
  --segment_threshold=0.9 \
  --segment_ar_probability
```
![Screenshot 2024-01-25 160006](https://github.com/LBL-EESA/TECA/assets/8796694/42704b59-7ffb-44b7-9da8-56b8da5085f2)

This aligns with what I expect from the AR probability field:
![Screenshot 2024-01-25 155945](https://github.com/LBL-EESA/TECA/assets/8796694/bc9a268c-b1b0-4810-bfbd-249cdef28739)

I'll note that the warning messages are unlikely to be associated with this.  

Finally, now that I'm looking carefully, I see that it looks like you're using TECA to calculate IVT from U, V, and Q.  It can do this, but it will save you time (and be more consistent with ERA5) if you use the IVT values from ERA5: the vertically integrated moisture flux (east & north) values.

**Comment by VThandlam** [comment]
Hi @taobrienlbl
I wanted to thank you for your detailed response regarding my BARD issue. I was expecting similar results to yours, and your reply helped me to solve the problem. After a relook, I realized that I made a mistake in using "--segment_threshold". I used it as "-segment_threshold" in my runs. Once I corrected it, the desired results were outputted. Thank you very much for your help.

I noticed that you are using .mcf files, and you have integrated moisture flux values for east and north at a single level. I am using NetCDF data for Q, U, and V of individual months that I downloaded from CDS. I am testing different pressure level integrations.
